### PR TITLE
Document BSPv25

### PR DESCRIPTION
**Pull request by JJL772**
Documents the BSP v25 format as extensively as possible. Anything not changed over v21 has been omitted.

### CODE REVIEW

File: docs/Reference/bsp-v25.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+```c
+struct lumpfileheader_t
+{
+	int				lumpOffset;
+	int				lumpID;
+	int				lumpVersion;	// Added by BSP v25
+	int				lumpLength;
+	int				mapRevision;
+};
+```
```
Comment (SCell555) [inline docs/Reference/bsp-v25.md:89]: This always existed.

File: docs/Reference/bsp-v25.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+### Lump Versions
+
+The following table lists current lump versions for all lumps that are versioned in Chaos:
+
+| Lump Name | Version |
+|---|---|
+| `LUMP_LIGHTING_VERSION` | 1 |
+| `LUMP_OCCLUSION_VERSION` | 2 |
+| `LUMP_LEAF_AMBIENT_LIGHTING_VERSION`  1 |
+| `LUMP_WORLDLIGHTS_VERSION` | 1 |
```
Comment (SCell555) [inline docs/Reference/bsp-v25.md:100]: These 4 did not get changed from v21/20.

File: docs/Reference/bsp-v25.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+## Structures
+
+In the BSP v25 update, all lumps listed in the lump version table have been modified. 
+This section lists the new structure format for each of those lumps.
+
+```cpp
+struct dphysdisp_t
+{
+	unsigned int numDisplacements;
```
Comment (SCell555) [inline docs/Reference/bsp-v25.md:129]: Could be important to note that the ushort array that follows this header got changed to uint array. `//unsigned int dataSize[numDisplacements];`

File: docs/Reference/bspsupport.md [hunk 4]
```diff
@@ -28,6 +28,7 @@ For more details, see
...
 | Source  | 19-20      | HL2, TF2, CSS                                             | Yes       | -            | Fully supported                                                          |
 | Source  | 20         | GMod, L4D, Black Mesa                                     | Yes       | -            | Map loads, parts might be broken due to missing shaders/scripting, etc.  |
-| Source  | 21         | Portal 2, CSGO, Alien Swarm, any Chaos VBSP compiled maps | Yes       | -            | Natively supported, preferred for any Chaos game                         |
+| Source  | 21         | Portal 2, CSGO, Alien Swarm                               | Yes       | -            | Fully supported                                                          |
 | Source  | 21         | Left 4 Dead 2                                             | No        | Probably not | Can be fixed by swapping entries in lump headers around, see link above  |
 | Source  | ≥22        | pre-Reborn Dota 2, Contagion, any Respawn Source games    | No        | No           | Too many undocumented differences to make it worth                       |
```
Comment (SCell555) [inline docs/Reference/bspsupport.md:33]: Now we support bsp v22 that is used by Infra.

File: docs/Reference/bsp-v25.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+	int			m_iDispTriStart;
+    int         power;
+    int         minTess;
+    float       smoothingAngle;
+    int         contents;
+	unsigned int	m_iMapFace;
+	int			m_iLightmapAlphaStart;
+	int			m_iLightmapSamplePositionStart;
+	CDispNeighbor<unsigned int>			m_EdgeNeighbors[4];
+	CDispCornerNeighbors<unsigned int>	m_CornerNeighbors[4];
```
Comment (SCell555) [inline docs/Reference/bsp-v25.md:157]: CDispNeighbor and CDispCornerNeighbors are not originally template types, do we want to have here an equivalent non-template type here?

**Review by SCell555 — CHANGES_REQUESTED** [review]

File: docs/Reference/bsp-v25.md (hunk 3 above)
Comment (JJL772) [reply docs/Reference/bsp-v25.md:129]: oh okay, I just mindlessly removed that comment haha

File: docs/Reference/bsp-v25.md (hunk 1 above)
Comment (JJL772) [reply docs/Reference/bsp-v25.md:89]: Misread the git blame, oops!

File: docs/Reference/bsp-v25.md (hunk 5 above)
Comment (JJL772) [reply docs/Reference/bsp-v25.md]: Changed these to generic struct types now

File: docs/Reference/bsp-v25.md (hunk 3 above)
Comment (SCell555) [reply docs/Reference/bsp-v25.md:109]: Also that comment in our header is now wrong, it still says short there.

**Review by SCell555 — APPROVED** [review]

**Pull request merged**